workflow/executor/agent.go

```diff
@@ -179,36 +195,36 @@ func (ae *AgentExecutor) patchWorker(ctx context.Context, taskSetInterface v1alp
 			// Patch was successful, clear nodeResults for next iteration
 			nodeResults = map[string]wfv1.NodeResult{}
 
-			ae.log.Info("Patched TaskSet")
+			log.Info("Patched TaskSet")
 		}
 	}
 }
 
-func (ae *AgentExecutor) processTask(ctx context.Context, tmpl wfv1.Template) (*wfv1.NodeResult, error) {
+func (ae *AgentExecutor) processTask(ctx context.Context, tmpl wfv1.Template) (*wfv1.NodeResult, time.Duration, error) {
+	var executeTemplate templateExecutor
 	switch {
 	case tmpl.HTTP != nil:
-		var result wfv1.NodeResult
-		if outputs, err := ae.executeHTTPTemplate(ctx, tmpl); err != nil {
-			result.Phase = wfv1.NodeFailed
-			result.Message = err.Error()
-		} else {
-			result.Phase = wfv1.NodeSucceeded
-			result.Outputs = outputs
-		}
-		return &result, nil
+		executeTemplate = ae.executeHTTPTemplate
```

Comment:
Perhaps better is an `interface` pattern?

```go
type AgentExecutable interface {
  Execute(ctx context.Context, tmpl wfv1.Template, reply *wfv1.NodeResult) error
  Requeue() time.Duration
}

// ...

type HTTPAgentExectuable struct {
}

var _ AgentExecutable = &HTTPAgentExectuable{}

func (ae *HTTPAgentExectuable) Execute(ctx context.Context, tmpl wfv1.Template, reply *wfv1.NodeResult) error {
  // ...
}

func (ae *HTTPAgentExectuable) Requeue() time.Duration {
  return time.Duration(0)
}


// ...

type PluginAgentExectuable struct {
  requeue time.Duration
}

var _ PluginAgentExectuable = &HTTPAgentExectuable{}

func (ae *PluginAgentExectuable) Execute(ctx context.Context, tmpl wfv1.Template, reply *wfv1.NodeResult) error {
  // ...

  // somewhere in here
  ae.requeue = // some value

  // ...
}

func (ae *PluginAgentExectuable) Requeue() time.Duration {
  return ae.requeue
}
```

Reply:
I'm not sure the right solution for long running plugin templates. Currently, you return when to re-queue the request. I wanted to avoid a callback endpoint.

Reply:
This suggestion is purely a Go-lang implementation detail. No need for new endpoints/changes to the API here.